src/generators/pm-install.ts

```diff
@@ -0,0 +1,46 @@
+import { readPackageJSON } from "pkg-types";
+import { codeBlock } from "omark";
+import { defineGenerator } from "../generator";
+
+export default defineGenerator({
+  name: "pm-install",
+  async generate({ options, args }) {
+    const name = args.name || (await inferPackageName(options.dir));
+
+    if (!name) {
+      return {
+        contents: "<!-- package name is unspecified -->",
+      };
+    }
+
+    const pkgInstalls = [
+      ["npm", "install"],
+      ["yarn", "add"],
+      ["pnpm", "install"],
+      ["bun", "install"],
+    ];
+
+    if (args.auto ?? true) {
+      pkgInstalls.unshift(["npx nypm@latest", "add"]);
+    }
+
+    return {
+      contents: codeBlock(
+        pkgInstalls
+          .map(
+            ([cmd, install]) =>
+              `# ${cmd.includes("nypm") ? "✨ Auto-detect" : cmd}\n${cmd} ${install} ${name}${args.dev ? " -D" : ""}`,
```

Comment:
Adds support for global packages (`global="true"` usage), use `===` instead of includes for comparisons with `cmd`, move dev mode before package name.
```suggestion
              `# ${cmd === "nypm" ? "✨ Auto-detect" : cmd}\n${cmd} ${args.global && cmd === "yarn" ? "  global" : ""}${install}${args.dev ? " -D" : ""}${args.global && cmd !== "yarn" ? " -g" : ""} ${name}`,
```

Reply:
@uncenter Only reason for the `.includes` was to keep it from looking too ugly XD but I can add support for global.

Reply:
xD sounds good. I've confirmed that `pnpm`, `npm`, and `bun` support the `-g`/`--global` flags, and yarn requires the weird `yarn global add` command.

Reply:
Ooh `nypm` doesn't support a global flag though (https://github.com/unjs/nypm/blob/main/src/cli.ts)... `${args.global && ["yarn", "nypm"].includes(cmd) ? " -g" : ""}` in that case.

Reply:
Yeah global is dangerous 😓 (although PR is welcome to add it for nypm and we can still have are here. but I won't advocate for it personally..)

Reply:
What's the issue with global?

Reply:
- dependencies can conflict with each other (varies between how pms work in global mode)
- outdated binaries are around and even sometimes conflict with their project-local versions

Reply:
~> https://github.com/unjs/nypm/issues/112